placement/algo/helper.go

```diff
@@ -227,7 +229,7 @@ func (ph placementHelper) getShardLen() int {
 	return len(ph.uniqueShards)
 }
 
-func (ph placementHelper) canAssignRack(shard uint32, from placement.HostShards, toRack string) bool {
+func (ph placementHelper) CanMoveShardToRack(shard uint32, from placement.HostShards, toRack string) bool {
```

Comment:
Perhaps call this `HasNoRackConflict`? We might decide to move `shard` to `toRack` even if this function returns false if the rack constraint is loosened.